### ISSUE TITLE
Idea Hub CTA is hidden when you connect a service

### ISSUE THREAD

**Issue by wpdarren** [google/site-kit-wp#4274]
## Bug Description

Currently, when you connect Site Kit or another service, the top Idea Hub CTA is hidden behind the Setup CTA `Congrats on completing the setup for X` banner.  When you click on the dismiss link (OK, Got it!), go to another page and come back to Site Kit dashboard) the Idea Hub CTA appears.  

Ideally, when the user dismisses the set up CTA, the Idea Hub CTA should appear underneath when refreshing. 

https://app.asana.com/0/1200491083500938/1200758733177801

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* After dismissing the setup/module success notification in the Site Kit dashboard (and when the `ideaHubModule` feature flag is enabled), the Idea Hub prompt notification should appear in the Site Kit dashboard if the user refreshes the page.
* Technically this is currently prevented by the `notification=authentication_success` query parameter remaining in the current URL and thus blocking other notifications. To fix this, upon dismissal of that success notification, this query parameter should be cleared so that in case the user refreshes the current page, they land on the "plain" Site Kit dashboard URL without extra query parameters.

## Implementation Brief
* Using `assets/js/components/notifications/SetupSuccessBannerNotification.js`, update the function for the `onDismiss` prop for `BannerNotification` to clear the `notification` URL parameter.
* To get the updated URL without the `notification` parameter, we can use the `removeQueryArgs` function from `@wordpress/url` and passing the current URL (`global.location.href`) and `notification` as the arguments.
* Then use `global.history.replaceState` to update to the new URL, passing `null`, an empty string, `''` and the new URL without the notification URL parameter respectively to the function.

### Test Coverage
* No new tests to be added.

## QA Brief
* Reset Site Kit
* Connect a module and the success notification will be displayed. Notice the `notification` URL parameter.
* Dismiss the module by clicking the `OK, Got it!` button.
* The `notification` URL parameter should no longer be present in the URL.
* Refresh the page, and check if other notifications are displayed correctly.

## Changelog entry

* Fix issue with the Idea Hub notification banner that was blocked by the setup success banner.

**Comment by eugene-manuilov** [comment]
> Using assets/js/components/notifications/SetupSuccessBannerNotification.js, update the function for the onDismiss prop for BannerNotification to clear the notification URL param by using the removeQueryArgs function from @wordpress/url.

@asvinb I am not sure that it will help. The `removeQueryArgs` function removes query arguments from the URL passed into it. It doesn't change the current page URL. We probably need to use something like `history.replaceState` to replace the current url with a new one that doesn't have the `notification=authentication_success` part.

**Comment by asvinb** [comment]
@eugene-manuilov Apologies for that. I assumed it was kinda implied. I updated the IB to be more detailed.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@asvinb an observation and interested in your thoughts.   

Here are the steps to recreate this issue. 

1. I connect a module and the success notification is displayed. 
2. The notification URL parameter is `wp-admin/admin.php?page=googlesitekit-dashboard&notification=authentication_success`  
3. I then dismiss the module by clicking the `OK, Got it!` button.  
4. The notification URL parameter is no longer present in the URL. `wp-admin/admin.php?page=googlesitekit-dashboard` 

When I refresh the page, I am still unable to see the Idea Hub notification unless I scroll up the page. 

You can see the jump from the top of the page to the `Site Overview` section on refresh in this screencast. 

Please note: on this screencast I did not have Analytics connected but the issue still arises when connected.

https://user-images.githubusercontent.com/73545194/140957221-776433a2-852c-47fb-a613-c52fabeb83dd.mp4

**Comment by asvinb** [comment]
@wpdarren So the issue you found during QA is a separate issue. Previously, the notification would not rendered at all because of the `notification` URL parameter. So if you dismiss a notification and you refresh the page, you will see new notifications if there are. So this is fixed.
The issue is you're describing is more about the notification being rendered but the scroll position is being maintained upon page refresh. You will notice that if you copy and paste the same URL in a new tab, you won't have the issue. The thing is that when you refresh the page, browsers tend to restore the last scroll position. In our case, there was a notification which is no longer there (since it was dismissed) and this behaviour varies across browsers.
@aaemnnosttv I'am thinking how critical is this issue or we can leave it or maybe create another ticket which checks for possible solutions. We can possibly have the browser scrolled to top upon mount of `BannerNotification` but is it worth it. What do you think @aaemnnosttv ?

**Comment by felixarntz** [comment]
@asvinb @wpdarren @aaemnnosttv Hmm, the point raised here doesn't seem like a critical issue to me. Unless I'm missing something, I'm not sure we need to fix something here at all, since maintaining the scroll position on refresh is default browser behavior. 🤔

**Comment by aaemnnosttv** [comment]
@felixarntz the weird thing is that the module setup success notification is shown at the top of the page. When scrolled to the top of the page and dismissing that notification and then refreshing the page, the Idea Hub notification is then hidden by scroll which seems to have been added rather than sitting at the top of the window as expected.

https://user-images.githubusercontent.com/1621608/141852852-1b430925-6620-4cd8-94ed-e9be9a9f906c.mp4

**Comment by felixarntz** [comment]
@aaemnnosttv Ah thanks, that is weird indeed. It may be that the fix for it is very simple, but it might also be a bit obscure to find a proper solution. Maybe we create a follow-up issue for this @wpdarren @asvinb?

**Comment by asvinb** [comment]
@felixarntz Yes, I agree. It'll require some digging to find a solution for this issue. Follow up issue created here:
https://github.com/google/site-kit-wp/issues/4356

@wpdarren @felixarntz @aaemnnosttv It's also good to note that I was not able to reproduce the scrolling bug on the unified dashboard, which eventually can be a non issue once we roll unified dashboard.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

- After connecting a module and the success notification is displayed, the notification URL parameter is.  `wp-admin/admin.php?page=googlesitekit-dashboard&notification=authentication_success&slug=analytics`. When you dismiss the notification by clicking the `OK, Got it!` button, the notification URL parameter should no longer be present in the URL. `wp-admin/admin.php?page=googlesitekit-dashboard&slug=analytics`

- When refreshing the page, the Idea hub notifications is displayed correctly.
- Also checked the same on smaller screen sizes. 

![image](https://user-images.githubusercontent.com/73545194/141994451-01ad6c31-3424-4c21-9365-144e22777c74.png)
